DeepSpeech.py

```diff
@@ -3,179 +3,168 @@
-
-# Learning rate parameter
-learning_rate = float(os.environ.get('ds_learning_rate', 0.001))
-
+tf.app.flags.DEFINE_float   ('beta1',            0.9,         'Beta 1 parameter of Adam optimizer')
+tf.app.flags.DEFINE_float   ('beta2',            0.999,       'Beta 2 parameter of Adam optimizer')
+tf.app.flags.DEFINE_float   ('epsilon',          1e-8,        'Epsilon parameter of Adam optimizer')
+tf.app.flags.DEFINE_float   ('learning_rate',    0.001,       'Learning Rate of Adam optimizer')
 
 # Batch sizes
 
-# The number of elements in a training batch
-train_batch_size = int(os.environ.get('ds_train_batch_size', 1))
-
-# The number of elements in a dev batch
-dev_batch_size = int(os.environ.get('ds_dev_batch_size', 1))
-
-# The number of elements in a test batch
-test_batch_size = int(os.environ.get('ds_test_batch_size', 1))
-
+tf.app.flags.DEFINE_integer ('train_batch_size', 1,           'number of elements in a training batch')
+tf.app.flags.DEFINE_integer ('dev_batch_size',   1,           'number of elements in a validation batch')
+tf.app.flags.DEFINE_integer ('test_batch_size',  1,           'number of elements in a test batch')
 
 # Sample limits
 
-# The maximum amount of samples taken from (the beginning of) the train set - 0 meaning no limit
-limit_train = int(os.environ.get('ds_limit_train', 0))
-
-# The maximum amount of samples taken from (the beginning of) the validation set - 0 meaning no limit
-limit_dev   = int(os.environ.get('ds_limit_dev',   0))
-
-# The maximum amount of samples taken from (the beginning of) the test set - 0 meaning no limit
-limit_test  = int(os.environ.get('ds_limit_test',  0))
-
+tf.app.flags.DEFINE_integer ('limit_train',      0,           'maximum number of elements to use from train set')
+tf.app.flags.DEFINE_integer ('limit_dev',        0,           'maximum number of elements to use from validation set')
+tf.app.flags.DEFINE_integer ('limit_test',       0,           'maximum number of elements to use from test set')
 
 # Step widths
 
-# The number of epochs we cycle through before displaying progress
-display_step = int(os.environ.get('ds_display_step', 1))
-
-# The number of epochs we cycle through before checkpointing the model
-checkpoint_step = int(os.environ.get('ds_checkpoint_step', 5))
-
-# The number of epochs we cycle through before validating the model
-validation_step = int(os.environ.get('ds_validation_step', 0))
-
+tf.app.flags.DEFINE_integer ('display_step',     1,           'number of epochs we cycle through before displaying progress')
+tf.app.flags.DEFINE_integer ('validation_step',  0,           'number of epochs we cycle through before validating the model')
 
 # Checkpointing
 
-# The directory in which checkpoints are stored
-checkpoint_dir = os.environ.get('ds_checkpoint_dir', xdg.save_data_path('deepspeech'))
+tf.app.flags.DEFINE_string  ('checkpoint_dir',   '',          'directory in which checkpoints are stored')
```

Comment:
Could you document the directory used if this is not set, i.e. the default value of this.